drivers/flash/soc_flash_nrf.c

```diff
@@ -94,32 +94,24 @@ static inline bool is_aligned_32(uint32_t data)
 	return (data & 0x3) ? false : true;
 }
 
-static inline bool is_regular_addr_valid(off_t addr, size_t len)
+static inline bool is_within_bounds(off_t addr, size_t len, off_t boundary_start,
+				    size_t boundary_size)
 {
-	size_t flash_size = nrfx_nvmc_flash_size_get();
-
-	if (addr >= flash_size ||
-	    addr < 0 ||
-	    len > flash_size ||
-	    (addr) + len > flash_size) {
-		return false;
-	}
-
-	return true;
+	return !(addr >= boundary_start + boundary_size ||
+			 addr < boundary_start ||
+			 len > boundary_size ||
+			 addr + len > boundary_start + boundary_size);
 }
 
+static inline bool is_regular_addr_valid(off_t addr, size_t len)
+{
+	return is_within_bounds(addr, len, 0, nrfx_nvmc_flash_size_get());
+}
 
 static inline bool is_uicr_addr_valid(off_t addr, size_t len)
 {
 #ifdef CONFIG_SOC_FLASH_NRF_UICR
-	if (addr >= (off_t)NRF_UICR + sizeof(*NRF_UICR) ||
```

Comment:
de-referencing NRF_UICR before taking it's size is a bit suspicous. Won't that give us the first word of the UICR, so 4 bytes ?

Reply:
Tested this. sizeof(*NRF_UICR) is correct.